map/framework.cpp

```diff
@@ -2433,23 +2433,20 @@ std::string Framework::GetMapLanguageCode()
   return languages::GetCurrentMapLanguage();
 }
 
-void Framework::SetMapLanguageCode(std::string const & languageCode)
+void Framework::SetMapLanguageCode(std::string const & langCode)
 {
-  settings::Set(settings::kMapLanguageCode, languageCode);
-  if (m_drapeEngine == nullptr)
-    return;
+  settings::Set(settings::kMapLanguageCode, langCode);
+  if (m_drapeEngine)
+    ApplyMapLanguageCode(langCode);
+}
 
-  int8_t const languageCodeIndex = [&languageCode]()
-  {
-    if (languageCode.empty())
-      return StringUtf8Multilang::kDefaultCode;
-    int8_t const index = StringUtf8Multilang::GetLangIndex(languageCode);
-    if (index == StringUtf8Multilang::kUnsupportedLanguageCode)
-      return StringUtf8Multilang::kDefaultCode;
-    return index;
-  }();
+void Framework::ApplyMapLanguageCode(std::string const & langCode)
+{
+  int8_t langIndex = StringUtf8Multilang::GetLangIndex(langCode);
+  if (langIndex == StringUtf8Multilang::kUnsupportedLanguageCode)
```

Comment:
The list of available map languages should be fixed/hard-coded, and user should not be allowed to select something unsupported (otherwise we'll get a lot of support requests). How come that this function can receive an unsupported language?

Reply:
Safety check, added assertion.

Reply:
Maybe `LOG(LERROR, ...)` (or CRITICAL) would be better? Then users can report it even from release builds.